### ISSUE TITLE
No variables in variable explorer

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#406]
_From [franap...@gmail.com](https://code.google.com/u/115562478474757105196/) on 2010-10-07T12:32:57Z_

1. Create a variable in any console (internal or not).
2. Do you see it in the variable explorer? I don't

v2 beta4 on Ubuntu Lucid 10.04 x86_64

I've been working with v1 for a while (thanks a lot for it!) and the variable explorer was working fine. Am I the only one with this issue?

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=406_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-10-07T13:15:49Z_

Could you please check that you are not trying to show an "excluded" variable?
(see the excluded filters in the Variable Explorer options menu)

For example, upper-cased ref names are by default excluded.

The simplest example that should work is entering:
a = 1
The integer 'a' should be shown in the variable explorer, after the automatic refresh timeout (3000ms by default). If it still does not work, try to show the internal console to see if there is no traceback there.

**Status:** Accepted  
  **Labels:** -Priority-Medium Priority-High

**Comment by spyder-issues-migrator** [comment]
_From [franap...@gmail.com](https://code.google.com/u/115562478474757105196/) on 2010-10-08T03:13:13Z_

Thanks Pierre.
Yes, I confirm that if I enter "a=1" I don't see anything in the variable explorer, even removing all the filters and clicking "refresh". I don't see anything in the internal console neither. Let me know if I can help more.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-10-08T09:40:51Z_

What versions of PyQt and IPython are you using?
Is it happening in both IPython and Python interpreters?

BTW, it is normal that nothing appears in the variable explorer for the internal console: in Spyder v2.0, the internal console is just intended for internal debugging.

**Comment by spyder-issues-migrator** [comment]
_From [franap...@gmail.com](https://code.google.com/u/115562478474757105196/) on 2010-10-08T10:17:34Z_

Python 2.6.5, Qt 4.6.2, PyQt 4.7.2
I installed IPython to try: same result.
I just remembered that I made no "setup.py install" thing: is this supposed to make any difference? Everything but the var. exp. works fine.

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-10-08T10:21:48Z_

Yes it does make a difference... like disable features such as variable explorer and code introspection features in console!!

As it is mentioned on the installation page, either do one of the following:
  * installation: python setup.py install
  * add the directory containing the 'spyderlib' package to your PYTHONPATH environment variable

**Comment by spyder-issues-migrator** [comment]
_From [tim.mich...@gmail.com](https://code.google.com/u/117399810383525443135/) on 2010-10-08T10:27:51Z_

Hello,
I have not such a probelm on Ubuntu.

Please check this package out: https://edge.launchpad.net/~pythonxy/+archive/individual-devel

**Comment by spyder-issues-migrator** [comment]
_From [franap...@gmail.com](https://code.google.com/u/115562478474757105196/) on 2010-10-09T06:27:01Z_

Ok, sorry for not thinking about it before. I did follow the README in the first place, but the installation failed complaining about a missing directory and wouldn't accept the --install-dir parameter (I'm writing by memory now). So I tried to run the script as is and it seemed to run fine.

The DEB package works flawlessly on this machine (it's another one). I will try it on the original machine on Monday. Thank you.
